module_utils/network_lsr/ethtool.py

```diff
@@ -0,0 +1,43 @@
+# SPDX-License-Identifier: BSD-3-Clause
+
+import array
+import struct
+import fcntl
+import socket
+
+from .utils import Util
+
+ETHTOOL_GPERMADDR = 0x00000020
+SIOCETHTOOL = 0x8946
+
+
+def get_perm_addr(ifname):
+    """
+    Return the Permanent address value for the specified interface using the
+    ETHTOOL_GPERMADDR ioctl command.
+
+    Please for further documentation, see:
+    https://github.com/torvalds/linux/blob/master/include/uapi/linux/ethtool.h#L734
+    https://github.com/torvalds/linux/blob/master/include/uapi/linux/ethtool.h#L1388
+    https://git.kernel.org/pub/scm/network/ethtool/ethtool.git/tree/ethtool.c#n4172
+    """
+    try:
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sockfd = sock.fileno()
+        ifname = ifname.encode("utf-8")
+        ecmd = array.array("B", struct.pack("II6s", ETHTOOL_GPERMADDR, 6, b"\x00" * 6))
+        ifreq = struct.pack("16sP", ifname, ecmd.buffer_info()[0])
```

Comment:
struct will only use 16 bytes from ifname but ifname might be bigger in theory. Then it might return the wrong MAC address. How about checking the length of ifname after encoding and returning None if it is too long?

Reply:
Done thanks!